go-runtime-mixin/dashboards/go-runtime.json

```diff
@@ -0,0 +1,1152 @@
+{
+  "__inputs": [],
+  "__requires": [
+    {
+      "type": "grafana",
+      "id": "grafana",
+      "name": "Grafana",
+      "version": "7.2.0"
+    },
+    {
+      "type": "panel",
+      "id": "graph",
+      "name": "Graph",
+      "version": ""
+    },
+    {
+      "type": "datasource",
+      "id": "prometheus",
+      "name": "Prometheus",
+      "version": "1.0.0"
+    }
+  ],
+  "annotations": {
+    "list": [
+      {
+        "builtIn": 1,
+        "datasource": "-- Grafana --",
+        "enable": true,
+        "hide": true,
+        "iconColor": "rgba(0, 211, 255, 1)",
+        "name": "Annotations & Alerts",
+        "type": "dashboard"
+      }
+    ]
+  },
+  "description": "Go runtime metrics",
+  "editable": true,
+  "graphTooltip": 0,
+  "id": null,
+  "iteration": 1602794777869,
+  "links": [],
+  "panels": [
+    {
+      "aliasColors": {},
+      "bars": false,
+      "dashLength": 10,
+      "dashes": false,
+      "datasource": "$datasource",
+      "description": "Total bytes of memory reserved by the process.",
+      "fieldConfig": {
+        "defaults": {
+          "custom": {}
+        },
+        "overrides": []
+      },
+      "fill": 1,
+      "fillGradient": 0,
+      "gridPos": {
+        "h": 8,
+        "w": 12,
+        "x": 0,
+        "y": 0
+      },
+      "hiddenSeries": false,
+      "id": 16,
+      "legend": {
+        "avg": false,
+        "current": false,
+        "max": false,
+        "min": false,
+        "show": true,
+        "total": false,
+        "values": false
+      },
+      "lines": true,
+      "linewidth": 1,
+      "links": [],
+      "nullPointMode": "null",
+      "options": {
+        "dataLinks": []
+      },
+      "percentage": false,
+      "pointradius": 2,
+      "points": false,
+      "renderer": "flot",
+      "seriesOverrides": [],
+      "spaceLength": 10,
+      "stack": false,
+      "steppedLine": false,
+      "targets": [
+        {
+          "expr": "go_memstats_sys_bytes{job=\"$job\", instance=~\"$instance\"}",
```

Comment:
This will give us a line per selected job / instance - and therefore when there are many instances of a given job, it will give you many lines.  One rule of thumb I have is never have more than ~5 lines per graph.

In this case I would do a `sum(...)` here, and only show the total for all jobs.  Ditto other graphs.

Reply:
From the [original PR](https://github.com/prometheus/client_golang/pull/808):

> The panels in the dashboard apply no aggregations. I waffled on this a bit, but opted for that approach on the premise that the dashboard is basic enough, and the panels show few enough individual metrics, that the dashboard itself is most likely to be helpful for folks really just getting started. (That, and, they can use the job/instance template variables to drill into a particular instance if they really need to see something that's stuck.) Very open to persuasion on changing this to an aggregated approach.

But in retrospect, i think my reasoning should have led to the opposite conclusion - apply aggregations, on the premise that the user can drill into the individual instance, at which point the aggregation becomes a no-op.

Will change.

Reply:
(Oh, right. i was also trying to avoid the headache of deciding which aggregation was appropriate for the different metrics.)

Reply:
Yeah, but thats part of the fun of building these mixins!

Reply:
That is...literally true.

This is now revised, but i did avg, not sum. As the SUO here is the individual Go process, IMO the goal is to understand the behavior of each process, relative to one another; summing would be more about a fleet-level view of resource consumption.